### PR TITLE
Update build to handle docs release 0.26

### LINKED ISSUE #4275: Release 0.26
Tasks
- Update website to have new version 0.26, and drop oldest version
- Publish blog post with release notes for new version 0.26
    - Make sure to mention the first release of the component https://github.com/knative-sandbox/eventing-kogito

/assign

### PR DESCRIPTION
**Pull request by csantanapr**
Fixes #4275

### CODE REVIEW

**Comment by evankanderson** [comment]
I think you want to delete the `archived/v0.22-docs` folder like I did in #4289

File: hack/build.sh [hunk 1]
```diff
@@ -19,11 +19,11 @@ set -x
...
-VERSIONS_GENERATORS=("mkdocs" "mkdocs" "hugo" "hugo")  # update this to always be 4 in the next two releases replace hugo with mkdocs, remove the copy of static hugo site at the bottom
-RELEASE_BRANCHES=("v0.25.0" "v0.24.0")                     # Release version for serving/eventing yaml files and api references.
+VERSIONS=("0.26" "0.25" "0.24" "0.23")                  # Docs version, results in the url e.g. knative.dev/docs-0.23/..
+VERSIONS_GENERATORS=("mkdocs" "mkdocs" "mkdocs" "hugo")  # update this to always be 4 in the next two releases replace hugo with mkdocs, remove the copy of static hugo site at the bottom
+RELEASE_BRANCHES=("v0.26.0" "v0.25.0" "v0.24.0")                     # Release version for serving/eventing yaml files and api references.
 # 3) For now, set branches and repos for old versions of docs. (This will go away when all docs branches are release-$version).
-DOCS_BRANCHES=("release-0.25" "release-0.24") # add a branch here for the next 2 releases until everything is mkdocs
+DOCS_BRANCHES=("release-0.26" "release-0.25" "release-0.24") # add a branch here for the next 2 releases until everything is mkdocs
 REPOS=("knative" "knative" "knative")
```
Comment (evankanderson) [inline hack/build.sh:27]: I think you need to extend this list as well. See line 64 for where it is used.

**Comment by csantanapr** [comment]
>I think you want to delete the archived/v0.22-docs folder like I did in #4289

Good catch @evankanderson

**Comment by julz** [comment]
/retest

(now with less tests 🤞)

**Comment by evankanderson** [comment]
(This is already 20 successful, 1 pending, and 2 neutral checks, so it's ready for an `/approve` and a `/lgtm`.)

**Comment by csantanapr** [comment]
/hold
Lets wait for kn quickstart to install 0.26 https://github.com/knative-sandbox/homebrew-kn-plugins/pull/41

**Comment by csantanapr** [comment]
/unhold
Now `kn quickstart` `0.4.1+` works and it installs serving and eventing `0.26`

**Pull request merged**